Analysis/src/SignalObjectSelector.cpp

```diff
@@ -109,6 +129,47 @@ bool SignalObjectSelector::PassHTT_LeptonSelection(const ntuple::TupleLepton& le
     return true;
 }
 
+bool SignalObjectSelector::PassTauPOG_LeptonSelection(const ntuple::TupleLepton& lepton, Channel channel) const
+{
+    //againstElectron first, againstMuon second
+    static const std::map<Channel,std::pair<DiscriminatorWP,DiscriminatorWP>> againstDiscriminators =
+        { {Channel::ETau,{DiscriminatorWP::Tight,DiscriminatorWP::Loose}} ,
+          {Channel::MuTau,{DiscriminatorWP::VLoose,DiscriminatorWP::Tight}},
+          {Channel::TauTau,{DiscriminatorWP::VLoose,DiscriminatorWP::Loose}}
+      };
+
+    // WP for deepTau vs E (first) and deepTau vs Mu (second)
+    static const std::map<Channel,std::pair<DiscriminatorWP,DiscriminatorWP>> deepTauDiscriminators =
+        { {Channel::ETau,{DiscriminatorWP::VVVLoose,DiscriminatorWP::VVVLoose}} ,
+          {Channel::MuTau,{DiscriminatorWP::VVVLoose,DiscriminatorWP::VVVLoose}},
+          {Channel::TauTau,{DiscriminatorWP::VVVLoose,DiscriminatorWP::VVVLoose}}
+      };
+
+    static const std::map<Channel,double> pt_map =
+        { {Channel::ETau, cuts::H_tautau_2016::ETau::tauID::pt} ,
+          {Channel::MuTau, cuts::H_tautau_2016::MuTau::tauID::pt},
+          {Channel::TauTau, cuts::H_tautau_2016::TauTau::tauID::pt}
+      };
+
+    if(lepton.leg_type() == LegType::e) return true;
+    if(lepton.leg_type() == LegType::mu) {
+        if(!(lepton.p4().pt() > cuts::H_tautau_2017::MuTau::muonID::pt)) return false; //to be back
+        //if(!(lepton.iso() < cuts::H_tautau_2016::MuTau::muonID::pfRelIso04)) return false;
+        return true;
+    }
+    if(!(lepton.leg_type() == LegType::tau)) throw analysis::exception("Leg Type Default Selection not supported");
+    if(!(lepton.p4().pt() > pt_map.at(channel))) return false;
+    //if(lepton.decayMode() == 5 || lepton.decayMode() == 6 || lepton.decayMode() == 11) return false;
+    if(!(lepton.PassedOldDecayMode())) return false;
```

Comment:
Passed old decay mode should be applied only with TauPOG_default and TauPOG_dpfTau.
For other modes, only exclude dm 5 & 6.